### PR TITLE
Campaign fixes

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
**Base rescue mission specific**

- Fixes the ghost role not being able to free itself properly
- Fixed it not properly despawning at mission end
- Fixed the disk computers being incorrectly flagged as objectives

**General**

- Neutral faction can now respawn freely between missions (i.e. you're not locked out for 30 minutes if you went valhalla)
## Why It's Good For The Game
Fixes good.
## Changelog
:cl:
fix: fixed a few campaign bugs
qol: 3rd party faction are not locked behind the 30 min respawn timer in campaign, when between missions
/:cl:

### CODE REVIEW

File: code/datums/gamemodes/campaign/missions/base_rescue.dm [hunk 1]
```diff
@@ -145,6 +145,11 @@
 	else
 		icon_state = "[initial(icon_state)]_open"
 
+/obj/structure/weapon_x_pod/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
+	if(X.loc != src)
```
Comment (psykzz) [inline code/datums/gamemodes/campaign/missions/base_rescue.dm:149]:
```suggestion
	if(X != occupant)
	    return
```

This feels like it describes the check better, making things simpler to understand

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/hunter.dm [hunk 2]
```diff
@@ -24,3 +24,13 @@
 
 /mob/living/carbon/xenomorph/hunter/weapon_x
 	caste_base_type = /mob/living/carbon/xenomorph/hunter/weapon_x
+
+/mob/living/carbon/xenomorph/hunter/weapon_x/Initialize(mapload)
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_CAMPAIGN_MISSION_ENDED, PROC_REF(terminate_specimen))
```
Comment (psykzz) [inline code/modules/mob/living/carbon/xenomorph/castes/hunter/hunter.dm:30]: Make sure you clean up the signal in the Destroy proc

Comment (Lumipharon) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/hunter.dm:30]: are sigs not all cleaned up when destroyed anyway?

Comment (Arturlang) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/hunter.dm:30]: Any reason not to just apply the signal on when the hunter is spawned via the xeno pod, or is there a reason to do it on the mob itself via a separate hunter type.

Comment (Lumipharon) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/hunter.dm:30]: what

Comment (Arturlang) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/hunter.dm:30]: Ehh nevermind it wouldn't work anyways, sorry

Comment (ivanmixo) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/hunter.dm:30]: The top parent of destroys handles signals, you don't need to manually unregister.

https://github.com/tgstation/TerraGov-Marine-Corps/blob/949bc7a8a7bf4b69c67c8cf0d491053705b93733/code/datums/datum.dm#L125

**Review by ivanmixo — APPROVED** [review]

**Pull request merged**